zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -248,6 +249,138 @@ def _get_services(cls):
+class NeutronGatewayStatusActionsTest(test_utils.OpenStackBaseTest):
+    """Test status actions of Neutron Gateway Charm.
+
+    actions:
+      * get-status-routers
+      * get-status-dhcp
+      * get-status-lb
+    """
+
+    SKIP_LBAAS_TESTS = True
+
+    @classmethod
+    def setUpClass(cls, application_name='neutron-gateway', model_alias=None):
+        """Run class setup for running Neutron Gateway tests."""
+        super(NeutronGatewayStatusActionsTest, cls).setUpClass(
+            application_name, model_alias)
+        # set up clients
+        cls.neutron_client = (
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+        # Loadbalancer tests not supported on Train and above and on
+        # releases Mitaka and below
+        current_release = openstack_utils.get_os_release()
+        bionic_train = openstack_utils.get_os_release('bionic_train')
+        xenial_mitaka = openstack_utils.get_os_release('xenial_mitaka')
+        cls.SKIP_LBAAS_TESTS = (xenial_mitaka <= current_release or
+                                current_release >= bionic_train)
+
+    def tearDown(self):
+        """Cleanup loadbalancers if there are any left over."""
+        super(NeutronGatewayStatusActionsTest, self).tearDown()
+        if not self.SKIP_LBAAS_TESTS:
+            load_balancers = self.neutron_client.list_loadbalancers().get(
+                'loadbalancers', [])
+            for lbaas in load_balancers:
+                self.neutron_client.delete_loadbalancer(lbaas['id'])
+
+    def _assert_result_match(self, action_result, resource_list,
+                             resource_name):
+        """Assert that action_result contains same data as resource_list."""
+        # make sure that action completed successfully
+        self.assertEqual(action_result.status, 'completed')
+
+        # extract data from juju action
+        action_data = action_result.data.get('results', {}).get(resource_name)
+        resources_from_action = json.loads(action_data)
+
+        # pull resource IDs from expected resource list and juju action data
+        expected_resource_ids = {resource['id'] for resource in resource_list}
+        result_resource_ids = {resource['id'] for resource in
+                               resources_from_action}
+
+        # assert that juju action returned expected resources
+        self.assertEqual(result_resource_ids, expected_resource_ids)
+
+    def test_get_status_routers(self):
+        """Test that get-status-routers reports correct neutron routers."""
+        # fetch neutron routers using neutron client
+        ngw_unit = zaza.model.get_units(self.application_name,
+                                        model_name=self.model_name)[0]
+        routers_from_client = self.neutron_client.list_routers().get(
+            'routers', [])
+
+        if not routers_from_client:
+            self.fail('At least one router must be configured for this test '
+                      'to pass.')
```

Comment:
Do you want the test to fail, or just to be skipped?

Reply:
Intention was for tests to fail because the model is not properly configured. Unless there's a reason for some test models to not have routers. But none of the func test models had problem with this so far.